### ISSUE TITLE
Fix 2 way level transitions/stop forcing "new level unit" enabled

### ISSUE THREAD

**Issue by vrad-exe** [StrataSource/Portal-2-Community-Edition#56]
Upon transitioning back to a previous map using HL2-style changelevel triggers, all entities will be reset to their initial state. This does not happen in the exact same map in Portal 1. The description of the "New Level Unit" keyvalue in worldspawn indicates it would have been used to control this behavior, but it doesn't seem to be implemented in P2's engine, instead always being treated as on.

This should probably have a BSP version check so it doesn't apply to old maps, and possibly make the keyvalue default to enabled in the FGD since most maps won't need this (or make it a different keyvalue entirely).

**Comment by Luke18033** [comment]
I'd consider this more of a bug, since it used to work in older games.

**Comment by Luke18033** [comment]
Seems like Portal 2 just forces every transition to be treated as one-way, all the code should still be there.

**Comment by novaplusplus** [comment]
I recently ran into this when experimenting with seamless level transitions. I made a more specific test setup with HL2, specifically at the transition between d2_coast_03 and d2_coast_04. In the current version of half-life 2, I loaded d2_coast_04 then went back to _03, went back to one of the buildings (the one with the headcrab shell inside) far away from the level change trigger. I smashed the two crates and tossed the barrel away, then went back to the level change trigger, through it to _04, forward a ways, then turned around and went back through it to _03 again, and returned to the same building.

Original state:
![Image](https://github.com/user-attachments/assets/ee9cdc3a-7152-4522-a7b9-7990cb92a872)

After returning:
![Image](https://github.com/user-attachments/assets/612df492-de1d-4729-8833-ad9153628af5)

Crates are gone, gibs probably de-spawned before I made it to the trigger, and the barrel is where I left it.

After this I went and did the same test in P2CE with HL2 mounted:

Original state:
![Image](https://github.com/user-attachments/assets/15afbd49-28e9-4525-9531-e136ad722cf4)

After returning:
![Image](https://github.com/user-attachments/assets/e404b944-33d8-4007-9ddd-41d4c486811e)

(ignore the small bar near the crosshair, that's some unrelated panorama testing I was doing)

So in short: yeah, it doesn't work the same. There can hopefully be some method to allow both operation modes to exist for compatibility across games given Portal 2's different handling of this. But also for custom maps to be able to use seamless transitions without trouble.

And relatedly to that, here are a pair of test maps set up and compiled for P2CE for testing this scenario as well:

[changeleveltest.zip](https://github.com/user-attachments/files/25010498/changeleveltest.zip)
[changeleveltest_vmf.zip](https://github.com/user-attachments/files/25010499/changeleveltest_vmf.zip)
